### PR TITLE
[sdformat9] Add new port 🤖

### PR DESCRIPTION
**Pull request by traversaro**
- What does your PR fix? 
  - This PR adds support for install the sdformat9 library. 

- Which triplets are supported/not supported? Have you updated the CI baseline?
  - The port is not supported on UWP due to the use of Win32 functions not available in  UWP (see https://github.com/microsoft/vcpkg/pull/8137#issuecomment-531321156). Furthermore, it is not supported on Linux due to the malformed CMake config of one dependency (`urdfdom`).

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  - Yes

As for the closely related ignition libraries  (https://github.com/microsoft/vcpkg/pull/7781,  https://ignitionrobotics.org/) different major version of sdformat can be installed side by side, so each new major version is added as a new port. 

In particular, sdformat9 is a dependency for Gazebo 11 (http://gazebosim.org/blog/gazebo11) and for Ignition Robotics Citadel (that contains Ignition Gazebo 3, see https://www.openrobotics.org/blog/2019/12/11/ignition-citadel-released).

### CODE REVIEW

File: ports/sdformat9/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    SHA512 6fc7d0ce46d9a7d1cae4fd905ebe6a07bb4ba98faa267be92a32b3409d6d82a99d5082485008a15484f7b5be2c347b5b24bc472fb1a4be5eb8b678b105cae6af
+    HEAD_REF sdf9
+    # Backport of https://github.com/osrf/sdformat/pull/269
+    PATCHES respect-build-testing.patch
+)
+
+# Ruby is required by the sdformat build process
+vcpkg_find_acquire_program(RUBY)
+get_filename_component(RUBY_PATH ${RUBY} DIRECTORY)
+set(_path $ENV{PATH})
```
Comment (NancyLi1013) [inline ports/sdformat9/portfile.cmake:16]: Why do we need to add this here?

File: ports/sdformat9/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+vcpkg_add_to_path(${RUBY_PATH})
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS -DBUILD_TESTING=OFF
```
Comment (NancyLi1013) [inline ports/sdformat9/portfile.cmake:22]:
```suggestion
    OPTIONS 
        -DBUILD_TESTING=OFF
```

File: ports/sdformat9/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            -DUSE_EXTERNAL_URDF=ON
```
Comment (NancyLi1013) [inline ports/sdformat9/portfile.cmake:23]:
```suggestion
       -DUSE_EXTERNAL_URDF=ON
```

File: ports/sdformat9/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            -DUSE_EXTERNAL_TINYXML=ON
```
Comment (NancyLi1013) [inline ports/sdformat9/portfile.cmake:24]:
```suggestion
       -DUSE_EXTERNAL_TINYXML=ON
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/sdformat9/portfile.cmake (hunk 1 above)
Comment (traversaro) [reply ports/sdformat9/portfile.cmake:16]: It is used later (https://github.com/microsoft/vcpkg/pull/11265/files/e92e8c171b3e52fc0f4343958bd9a54c3cae6575#diff-0aa41acdc3ae5b35d9d840e2c97bd180R29) to restore the original path.

**Review by NancyLi1013 — APPROVED** [review]

File: ports/sdformat9/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [reply ports/sdformat9/portfile.cmake:16]: Thanks for your clarification.

**Comment by strega-nil** [comment]
Cool, thanks @traversaro!

**Pull request merged**